kong/core/handler.lua

```diff
@@ -119,9 +121,47 @@ return {
       end
 
       -- if set `host_header` is the original header to be preserved
-      var.upstream_host = host_header or 
+      var.upstream_host = host_header or
           balancer_address.hostname..":"..balancer_address.port
 
+      -- Keep-Alive and WebSocket Protocol Upgrade Headers
+      if var.http_upgrade == "websocket" then
```

Comment:
@thibaultcha, I pushed a patch to make case-insensitive comparison on Lua-land. I can always move it back, but I like this because it keeps our user facing `nginx.conf` cleaner.